src/main/kotlin/com/ampnet/payoutservice/controller/PayoutController.kt

```diff
@@ -1,45 +1,181 @@
+import com.ampnet.payoutservice.model.params.GetPayoutsForAdminParams
+import com.ampnet.payoutservice.model.params.GetPayoutsForInvestorParams
+import com.ampnet.payoutservice.model.result.FullCreatePayoutTask
+import com.ampnet.payoutservice.model.result.Payout
+import com.ampnet.payoutservice.repository.MerkleTreeRepository
 import com.ampnet.payoutservice.service.CreatePayoutQueueService
 import com.ampnet.payoutservice.util.BlockNumber
 import com.ampnet.payoutservice.util.ChainId
 import com.ampnet.payoutservice.util.ContractAddress
+import com.ampnet.payoutservice.util.PayoutStatus
+import com.ampnet.payoutservice.util.TaskStatus
 import com.ampnet.payoutservice.util.WalletAddress
+import mu.KLogging
 import org.springframework.http.ResponseEntity
 import org.springframework.security.core.annotation.AuthenticationPrincipal
 import org.springframework.web.bind.annotation.GetMapping
 import org.springframework.web.bind.annotation.PathVariable
 import org.springframework.web.bind.annotation.PostMapping
 import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestParam
 import org.springframework.web.bind.annotation.RestController
+import java.math.BigInteger
 import java.util.UUID
 
 @RestController
-class PayoutController(private val createPayoutQueueService: CreatePayoutQueueService) {
+class PayoutController(
+    private val createPayoutQueueService: CreatePayoutQueueService,
+    private val blockchainService: BlockchainService,
+    private val merkleTreeRepository: MerkleTreeRepository
+) {
 
-    @GetMapping("/payouts/tasks/{taskId}")
-    fun getPayoutTaskById(@PathVariable taskId: UUID): ResponseEntity<CreatePayoutTaskResponse> {
+    companion object : KLogging()
+
+    @GetMapping("/payouts/{chainId}/task/{taskId}")
+    fun getPayoutByTaskId(
+        @PathVariable chainId: Long,
+        @PathVariable taskId: UUID
+    ): ResponseEntity<PayoutResponse> {
+        logger.debug { "Get payout task by chainId: $chainId, taskId: $taskId" }
         return createPayoutQueueService.getTaskById(taskId)
-            ?.let { ResponseEntity.ok(it) }
+            ?.takeIf { it.chainId.value == chainId }
+            ?.let { ResponseEntity.ok(it.toPayoutResponse()) }
             ?: throw ResourceNotFoundException(
                 ErrorCode.PAYOUT_TASK_NOT_FOUND,
                 "Create payout task not found"
             )
     }
 
+    @Suppress("LongParameterList")
+    @GetMapping("/payouts/{chainId}")
+    fun getPayouts(
+        @PathVariable chainId: Long,
+        @RequestParam(required = false) issuer: String?,
+        @RequestParam(required = false) owner: String?,
+        @RequestParam(required = false) status: List<PayoutStatus>?,
+        @RequestParam(required = true) assetFactories: List<String>,
+        @RequestParam(required = true) payoutService: String,
+        @RequestParam(required = true) payoutManager: String
```

Comment:
I would put all mandatory/required params at the top, below chain id. It will be easier to read.